openedx/core/djangoapps/programs/signals.py

```diff
@@ -21,117 +23,59 @@
 
     Returns:
         None
-
     """
-    # Import here instead of top of file since this module gets imported before
-    # the credentials app is loaded, resulting in a Django deprecation warning.
-    from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
-
-    # Avoid scheduling new tasks if certification is disabled.
-    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
+    if not is_credentials_enabled():
         return
 
-    # schedule background task to process
     LOGGER.debug(
-        'handling COURSE_CERT_AWARDED: username=%s, course_key=%s, mode=%s, status=%s',
-        user,
-        course_key,
-        mode,
-        status,
+        f"Handling COURSE_CERT_AWARDED: user={user}, course_key={course_key}, mode={mode}, status={status}"
     )
     # import here, because signal is registered at startup, but items in tasks are not yet able to be loaded
     from openedx.core.djangoapps.programs.tasks import award_program_certificates
     award_program_certificates.delay(user.username)
 
 
 @receiver(COURSE_CERT_CHANGED)
-def handle_course_cert_changed(sender, user, course_key, mode, status, **kwargs):
+def handle_course_cert_changed(sender, user, course_key, mode, status, **kwargs):  # pylint: disable=unused-argument
     """
-        If a learner is awarded a course certificate,
-        schedule a celery task to process that course certificate
-
-        Args:
-            sender:
-                class of the object instance that sent this signal
-            user:
-                django.contrib.auth.User - the user to whom a cert was awarded
-            course_key:
-                refers to the course run for which the cert was awarded
-            mode:
-                mode / certificate type, e.g. "verified"
-            status:
-                "downloadable"
-
-        Returns:
-            None
+    When the system updates a course certificate, enqueue a celery task responsible for syncing this change in the
+    Credentials IDA
+
+    ***** Important *****
+    While the current name of the enqueue'd task is `award_course_certificate` it is *actually* responsible for both
+    awarding and revocation of course certificates in Credentials.
+    *********************
+
+    Args:
+        user(User): The user to whom a course certificate was awarded
+        course_key(CourseLocator): The course run key for which the course certificate was awarded
+        mode(str): The "mode" of the course (e.g. Audit, Honor, Verified, etc.)
+        status(str): The status of the course certificate that was awarded (e.g. "downloadable")
+
+    Returns:
+        None
     """
-    # Import here instead of top of file since this module gets imported before
-    # the credentials app is loaded, resulting in a Django deprecation warning.
-    from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
-
-    verbose = kwargs.get('verbose', False)
```

Comment:
is it appropriate to remove this verbose logging option?

Reply:
I was sorely tempted to remove it. Maybe in my followup PR I'll remove this too and make the logging default. This seems like useful information all of the time.

Reply:
but you did remove it, didn't you? Am I missing it? I don't see this in the new code.

Reply:
I'm still confused, I do think that this verbose log has been removed. I trust your judgment completely if you think it should be gone, I just want to make sure that it wasn't an accidental removal.

Reply:
OH. You're right!!! I misunderstood your original comment. I _did_ remove this by accident. I should have looked at this more in depth. Thanks for circling back around on this. I am going to re-add this log statement.